File: Java/TP3-3/src/Constelacao.java
import java.util.ArrayList;

public class Constelacao {
	private String nome;
	private ArrayList<Estrela> estrelas;

	public Constelacao(String nome) {
		this.nome = nome;
		this.estrelas = new ArrayList<>();
	}

	public void adicionar(Estrela e) {
		this.estrelas.add(e);
	}

	public void infoEstrelas() {
		for (Estrela e : estrelas) {
			e.mostrar();
			System.out.println("----------");
		}
	}

	public double tempConstelacao() {
		double temperaturaTotal = 0;
		for (Estrela e : estrelas) {
			temperaturaTotal += e.getTemperatura();
		}
		return temperaturaTotal;
	}

	public String getNome() {
		return nome;
	}
}

File: Kotlin/TPK3-3/src/Constelacao.kt
import java.util.ArrayList

class Constelacao(nome: String) {
	var nome: String
	private var estrelas: ArrayList<Estrela>

	init {
		this.nome = nome
		this.estrelas = ArrayList<Estrela>()
	}

	fun adicionar(e: Estrela) {
		this.estrelas.add(e)
	}

	fun infoEstrelas() {
		for (e in estrelas) {
			e.mostrar()
			println("----------")
		}
	}

	fun tempConstelacao(): Double {
		var temperaturaTotal = 0.0
		for (e in estrelas) {
			temperaturaTotal += e.getTemperatura()
		}
		return temperaturaTotal
	}
}